internal/checks/errorlatency.go

```diff
@@ -0,0 +1,494 @@
+}
+
+func (l *LatencyStats) RecordLatency(latency time.Duration) {
+	if latency >= l.threshold {
+		l.circuitBreaker.RecordFailure()
+	} else {
+		l.circuitBreaker.RecordSuccess()
+	}
+}
+
+func (l *LatencyStats) IsOpen() bool {
+	// TODO(polsar): Dedup with ErrorStats.IsOpen.
+	return l.circuitBreaker.IsOpen() || l.circuitBreaker.IsHalfOpen()
+}
+
+func (l *LatencyStats) GetThreshold() time.Duration {
+	return l.threshold
+}
+
+func NewLatencyStats(routingConfig *conf.RoutingConfig, method string) LatencyCircuitBreaker {
+	return &LatencyStats{
+		threshold: getLatencyThreshold(routingConfig, method),
+		circuitBreaker: NewCircuitBreaker(
+			conf.DefaultLatencyTooHighRate,
+			getDetectionWindow(routingConfig),
+			getBanWindow(routingConfig),
+		),
+	}
+}
+
+// NewCircuitBreaker abstracts away the rather complex circuitbreaker.Builder API.
+// https://pkg.go.dev/github.com/failsafe-go/failsafe-go/circuitbreaker
+// https://failsafe-go.dev/circuit-breaker/
+func NewCircuitBreaker(
+	errorRate float64,
+	detectionWindow time.Duration,
+	banWindow time.Duration,
+) circuitbreaker.CircuitBreaker[any] {
+	// TODO(polsar): Check that `0.0 < errorRate <= 1.0` holds.
+	return circuitbreaker.Builder[any]().
+		HandleResult(false). // The false return value of the wrapped call will be interpreted as a failure.
+		WithFailureRateThreshold(
+			uint(math.Floor(errorRate*PercentPerFrac)), // Minimum percentage of failed requests to open the breaker.
+			MinNumRequestsForRate,
+			detectionWindow,
+		).
+		WithDelay(banWindow).
+		Build()
+}
+
+type ErrorLatencyCheck struct {
+	client                       client.EthClient
+	Err                          error
+	clientGetter                 client.EthClientGetter
+	metricsContainer             *metrics.Container
+	logger                       *zap.Logger
+	upstreamConfig               *conf.UpstreamConfig
+	routingConfig                *conf.RoutingConfig
+	errorCircuitBreaker          ErrorCircuitBreaker
+	methodLatencyBreaker         map[string]LatencyCircuitBreaker // RPC method -> LatencyCircuitBreaker
+	lock                         sync.RWMutex
+	ShouldRunPassiveHealthChecks bool
+}
+
+func NewErrorLatencyChecker(
+	upstreamConfig *conf.UpstreamConfig,
+	routingConfig *conf.RoutingConfig,
+	clientGetter client.EthClientGetter,
+	metricsContainer *metrics.Container,
+	logger *zap.Logger,
+) types.ErrorLatencyChecker {
+	c := &ErrorLatencyCheck{
+		upstreamConfig:               upstreamConfig,
+		routingConfig:                routingConfig,
+		clientGetter:                 clientGetter,
+		metricsContainer:             metricsContainer,
+		logger:                       logger,
+		errorCircuitBreaker:          NewErrorStats(routingConfig),
+		methodLatencyBreaker:         make(map[string]LatencyCircuitBreaker),
+		ShouldRunPassiveHealthChecks: routingConfig.PassiveLatencyChecking && (routingConfig.Errors != nil || routingConfig.Latency != nil),
+	}
+
+	if err := c.InitializePassiveCheck(); err != nil {
```

Comment:
nit: 	check before initializing the passive health check, like we do on L181?

```
if !c.ShouldRunPassiveHealthChecks {
     return
}
````